src/lit-html.ts

```diff
@@ -613,14 +617,25 @@ export class TemplateInstance {
         }
       }
     }
-    if (this.template.svg) {
-      const svgElement = fragment.firstChild!;
-      fragment.removeChild(svgElement);
-      const nodes = svgElement.childNodes;
-      for (let i = 0; i < nodes.length; i++) {
-        fragment.appendChild(nodes.item(i));
-      }
-    }
     return fragment;
   }
 }
+
+/**
+ * Reparents nodes, starting from `startNode` (inclusive) to `endNode`
+ * (exclusive), into another container (could be the same container), before
+ * `beforeNode`. If `beforeNode` is null, it appends the nodes to the
+ * container.
+ */
+export const reparentNodes =
```

Comment:
Let's order the arguments to put optional last and use `?` to mark the optional args.

And 🚲🏠: since they're all `Node`s, we can shorten the names:

```ts
const reparentNodes = (container: Node, start: Node, end: Node|null = null, before?: Node) {
```

I added the default value for `end` so that if it's omitted, it's defaulted to `null` to work with `node !== endNode`. Accidentally passing `undefined` would have caused an infinite loop. `beforeNode` is safe as `undefined` though.

Reply:
`start` may be `null`, [though](https://github.com/PolymerLabs/lit-html/pull/164/files#diff-7548be97e171131880082c935a962f68R166).

> beforeNode is safe as undefined though.

[MDN](https://developer.mozilla.org/en-US/docs/Web/API/Node/insertBefore) has a specific warning about this.